test_runner/src/main/kotlin/ftl/reports/JUnitReport.kt

```diff
@@ -24,5 +21,23 @@ object JUnitReport : IReport {
         } else {
             write(matrices, output, args)
         }
+
+        result?.let {
+            uploadToGcStorage(it, args)
```

Comment:
Why not call `GcStorage.uploadCiJUnitXml` directly and pass `reportPath(matrices, args).let(::File).name` as fileName argument?. Is simpler, there is no needs for `override fun reportPath`, `private fun uploadToGcStorage` and `val fileName`.

Reply:
@jan-gogo you thinking about something like:
```
GcStorage.uploadCiJUnitXml(result, args, reportPath(matrices, args).let { ::File }.name)
```
I think we don't need execute reportPath because it's inside make additional work
check
```
  fun reportPath(matrices: MatrixMap, args: IArgs): String {
        val path = resolveLocalRunPath(matrices, args)
        return Paths.get(path, reportName() + extension).toString()
    }
```
we dont need  val path = resolveLocalRunPath(matrices, args)
so maybe make something like:
```
 private val fileName = "${reportName()}$extension"
    override fun run(matrices: MatrixMap, result: JUnitTestResult?, printToStdout: Boolean, args: IArgs) {
        if(result == null){
            return
        }
        val output = result.xmlToString()

        if (printToStdout) {
            print(output)
        } else {
            write(matrices, output, args)
        }

        GcStorage.uploadCiJUnitXml(result, args, fileName)
    }
```

or 

```
private fun createReportFileName() = "${reportName()}$extension"
    override fun run(matrices: MatrixMap, result: JUnitTestResult?, printToStdout: Boolean, args: IArgs) {
        if(result == null){
            return
        }
        val output = result.xmlToString()

        if (printToStdout) {
            print(output)
        } else {
            write(matrices, output, args)
        }

        GcStorage.uploadCiJUnitXml(result, args, createReportFileName())
    }
```
Tell me what do you think please.

Reply:
not
```
GcStorage.uploadCiJUnitXml(result, args, reportPath(matrices, args).let { ::File }.name)
```
but
```
GcStorage.uploadCiJUnitXml(result, args, reportPath(matrices, args).let(::File).name)
```
I know that it make additional operations, but it's small impact on performance, and gives less additional code to maintain.